app-backend/http4s-util/src/main/scala/com/rasterfoundry/http4s/xray/Segment.scala

```diff
@@ -12,8 +12,8 @@ final case class Segment[F[_]](name: String,
                                annotations: Map[String, String],
                                _type: Option[String],
                                http: Option[XrayHttp]) {
+  val ec2Config = if (Config.ec2.ec2Plugin.isEnabled) Map("ec2" -> Config.ec2.ec2Data) else Nil
   val aws: Map[String, collection.Map[String, String]] = Map(
-    "ec2" -> Config.ec2.ec2Data,
     "ecs" -> Config.ecs.ecsInstance
-  )
+  ) ++ ec2Config
```

Comment:
This will conditionally add the EC2 metadata to tracing segments.

> `true` if an environment inspection determines X-Ray is operating in the correct environment for this plugin OR if X-Ray cannot accurately determine if it's in this plugin's environment

See: https://docs.aws.amazon.com/xray-sdk-for-java/latest/javadoc/com/amazonaws/xray/plugins/EC2Plugin.html#isEnabled()